core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/core/SparqlBuilderTest.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) ${year} Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.sparqlbuilder.core;
+
+import static org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf.iri;
+
+import org.eclipse.rdf4j.model.vocabulary.DC;
+import org.eclipse.rdf4j.sparqlbuilder.constraint.Expressions;
+import org.eclipse.rdf4j.sparqlbuilder.core.query.Queries;
+import org.eclipse.rdf4j.sparqlbuilder.core.query.SelectQuery;
+import org.eclipse.rdf4j.sparqlbuilder.graphpattern.GraphPatternNotTriples;
+import org.eclipse.rdf4j.sparqlbuilder.graphpattern.GraphPatterns;
+import org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for precedence order of all operators in SPARQL builder Queries.
+ */
+public class SparqlBuilderTest {
+
+	protected SelectQuery query;
+	protected static final String EXAMPLE_ORG_NS = "https://example.org/ns#";
+	protected static final String DC_NS = DC.NAMESPACE;
+
+	@Before
+	public void before() {
+		query = Queries.SELECT();
+	}
+
+	@Test
+	public void testLogicalOperatorPrecedence1() {
+		Prefix dc = SparqlBuilder.prefix("dc", iri(DC_NS)), ns = SparqlBuilder.prefix("ns", iri(EXAMPLE_ORG_NS));
+		Variable title = SparqlBuilder.var("title"), price = SparqlBuilder.var("price"), x = SparqlBuilder.var("x");
+
+		GraphPatternNotTriples pricePattern = GraphPatterns.and(x.has(ns.iri("price"), price))
+				.filter(Expressions.or(Expressions.lt(price, Rdf.literalOf(20)),
+						Expressions.and(Expressions.lt(price, Rdf.literalOf(50)),
+								Expressions.gt(price, Rdf.literalOf(30)))))
+				.optional();
+		System.out.println(query);
```

Comment:
Could you remove the System.out.println ? Just so there is less output to scroll through when running these tests.